src/Arcus.Testing.Assert/Failure/AssertionException.cs

```diff
@@ -29,5 +30,17 @@ public AssertionException(string message) : base(message)
         public AssertionException(string message, Exception innerException) : base(message, innerException)
         {
         }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AssertionException" /> class with serialized data.
+        /// </summary>
+        /// <param name="info">The <see cref="SerializationInfo" /> that holds the serialized object data about the exception being thrown.</param>
+        /// <param name="context">The <see cref="StreamingContext" /> that contains contextual information about the source or destination.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="info" /> is <see langword="null" />.</exception>
+        /// <exception cref="SerializationException">Thrown when the class name is <see langword="null" /> or <see cref="Exception.HResult" /> is zero (0).</exception>
+        protected AssertionException(SerializationInfo info, StreamingContext context)
```

Comment:
I believe this approach has been made obsolete as from .NET 8

See also this documentation:
https://learn.microsoft.com/en-us/dotnet/api/system.exception.-ctor?view=net-8.0#system-exception-ctor(system-runtime-serialization-serializationinfo-system-runtime-serialization-streamingcontext)

Reply:
Ah, so the warning came from .NET 6. Okay, then it's indeed no need to add this, since we're moving away from .NET 6 soon. 👍 Thx!